src/safeposix/syscalls/net_calls.rs

```diff
@@ -249,34 +321,55 @@ impl Cage {
                                 Ok(a) => a,
                                 Err(e) => return e,
                             };
 
-                            let bindret = sockobj.0.bind(&localaddr);
-                            if bindret < 0 {
-                                match Errno::from_discriminant(interface::get_errno()) {
-                                    Ok(i) => {return syscall_error(i, "connect", "The libc call to bind within connect failed");},
-                                    Err(()) => panic!("Unknown errno value from socket bind within connect returned!"),
-                                };
+                            if let interface::GenSockaddr::Unix(_) = localaddr {
+                                self.bind_inner_socket(sockfdobj, &localaddr, false, &*sockobj);
+                            } else {
+                                let bindret = sockobj.0.bind(&localaddr);
+                                if bindret < 0 {
+                                    match Errno::from_discriminant(interface::get_errno()) {
+                                        Ok(i) => {return syscall_error(i, "connect", "The libc call to bind within connect failed");},
+                                        Err(()) => panic!("Unknown errno value from socket bind within connect returned!"),
+                                    };
+                                }
+                                sockfdobj.localaddr = Some(localaddr);
                             }
-
-                            sockfdobj.localaddr = Some(localaddr);
+                        } 
+                        
+                        if let interface::GenSockaddr::Unix(_) = remoteaddr {
+                            let path = remoteaddr.path().clone();
+                            let truepath = normpath(convpath(path), self);
+                            if !NET_METADATA.domain_socket_table.contains_key(&truepath) {return syscall_error(Errno::ECONNREFUSED, "connect", "The libc call to connect failed!");}
+                            remoteclone = NET_METADATA.domain_socket_table.get(&truepath).unwrap().clone();
+                            sockobj.0.set_blocking(); // unix domain sockets block on connect evne if nb, for now we fake them so set blocking and then unset after
                         };
 
-                        let connectret = sockobj.0.connect(remoteaddr);
+                        let mut inprogress = false;
+                        let connectret = sockobj.0.connect(&remoteclone);
                         if connectret < 0 {
                             match Errno::from_discriminant(interface::get_errno()) {
-                                Ok(i) => {return syscall_error(i, "connect", "The libc call to connect failed!");},
+                                Ok(i) => {
+                                    if i == Errno::EINPROGRESS { inprogress = true; }
```

Comment:
Here is where the comment below refers to

Reply:
It technically could be done up here but I think would be less readable and would require doubling some code.

Reply:
I wanted to lump the change in connection states together.